cluster/kubernetes/manifests.go

```diff
@@ -7,12 +7,36 @@ import (
 	"github.com/weaveworks/flux/resource"
 )
 
+// namespacer assigns namespaces to manifests that need it (or "" if
+// the manifest should not have a namespace.
+type namespacer interface {
+	correctNamespace(kresource.KubeManifest) error
```

Comment:
I am not sure I like the stateful interface. I would probably do something like:

```
type namespacer interface {
  ClusterNamespace(kresource.KubeManifest) (string, error)
}
```